docs/development/oidc.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Testing OIDC
+weight: 5
+---
+
+OIDC is a protocol that allows web applications to authenticate users through an identity provider and obtain basic profile information about the user.
+
+## Testing OIDC with Kubernetes API Server
+
+This guide will walk you through the process of testing OIDC (OpenID Connect) on a Minikube cluster using an Azure App Registration. OIDC allows you to authenticate and interact with the cluster using kubectl:
+
+### Create an Application on Azure
+
+- Create an application on Azure and specify the redirect URL as `http://localhost:8000`.
+- Generate a secret for the application under "Certificates and Secrets" and make note of its value.
+
+### Start Minikube
+
+- Start Minikube without enabling OIDC: `minikube start`. This is done to avoid RBAC (Role-Based Access Control) issues during startup. However, we can enable OIDC once the cluster is running.
+
+### SSH into Minikube
+
+- Once the Minikube cluster is started, SSH into it: `minikube ssh`.
+- Install the Vim text editor: `sudo apt-get update && sudo apt-get install vim`.
+
+### Edit the kube-apiserver Configuration
+
+- Edit the kube-apiserver configuration file using Vim: `sudo vi /etc/kubernetes/manifests/kube-apiserver.yaml`.
+- Add the following lines to the configuration file:
+
+```bash
+    - --oidc-issuer-url=https://sts.windows.net/<tenantID from application>/
+    - --oidc-client-id=<Application Client ID>
+    - --oidc-username-claim=email
+```
+
+- Save the changes to the configuration file and exit Vim.
+
+### Wait for API Server Restart
+
+- Wait for approximately 1 minute for the API server to restart and apply the new configuration.
+
+### Install Kubelogin
+
+- Install Kubelogin, which is a tool for testing Kubernetes authentication:
+
+```bash
+# Homebrew (macOS and Linux)
+brew install int128/kubelogin/kubelogin
+```
```

Comment:
I don't know of anyone who uses homebrew on Linux.
You can link to the kubelogin installation docs, and then say "Here's how to do it with Homebrew as an example:"